src/main/kotlin/cacophony/codegen/patterns/CacophonyPattern.kt

```diff
@@ -1,22 +1,215 @@
 package cacophony.codegen.patterns
 
 import cacophony.codegen.instructions.Instruction
-import cacophony.controlflow.CFGNode
-import cacophony.controlflow.Register
-import cacophony.controlflow.RegisterLabel
+import cacophony.controlflow.*
 
 class CacophonyPattern {
+    /*****************************
+     * CHAPTER 1. VALUE PATTERNS *
+     *****************************/
+
     abstract class BinaryOpPattern {
-        val lhsRegisterLabel = RegisterLabel()
-        val rhsRegisterLabel = RegisterLabel()
+        val lhsLabel = ValueLabel()
+        val rhsLabel = ValueLabel()
+    }
+
+    abstract class UnaryOpPattern {
+        val childLabel = ValueLabel()
+    }
+
+    // TODO: reduce boilerplate (typed DSL?)
```

Comment:
To give you some inspiration: https://github.com/mhorod/cacophony/blob/a0ffd9363123be86c472f62813b023707d15d1ad/src/test/kotlin/cacophony/TestCfgUtils.kt#L100

I think a great improvement would be to move/copy the node building methods from tests and add the ones that are missing, so we could write something like
```
val tree = registerSlot(lhsLabel) add mem(add(rbp), integer(8))
```